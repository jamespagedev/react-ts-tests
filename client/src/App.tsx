import React, {useState} from 'react';
import {DefaultButton} from './components/buttons';

function App() {
  const [strParentState, setStrParentState] = useState('');
  return (
    <div className="app">
      {/* Test Components Here */}
      <h3>strParentState: &quot;{strParentState}&quot;</h3>
      <DefaultButton btnText="Click Me" updateParentState={() => setStrParentState('Default Button')} />
    </div>
  );
}

export default App;
