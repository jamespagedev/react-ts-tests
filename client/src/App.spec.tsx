import React from 'react';
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

const btnText = 'Click Me';
const displayDefaultButtonTextBeforeClick = 'strParentState: ""';
const displayDefaultButtonTextAfterClick = 'strParentState: "Default Button"';
const mockUpdateParentStateFn = jest.fn();
function TestContainer(): JSX.Element {
  return <App />;
}

// ===========================================================================
// =============================== start tests ===============================
// ===========================================================================
describe('DefaultButton', () => {
  it('matches snapshot', () => {
    const {container} = render(<TestContainer />);
    expect(container).toMatchSnapshot();
  });
  it('default button click updates text', async () => {
    const testComponent = render(<TestContainer />);
    const defaultBtn = testComponent.getByRole('button', {name: btnText});
    const displayTextBeforeClick = testComponent.getByText(displayDefaultButtonTextBeforeClick);
    expect(displayTextBeforeClick).toBeInTheDocument();
    await userEvent.click(defaultBtn);
    const displayTextAfterClick = testComponent.getByText(displayDefaultButtonTextAfterClick);
    expect(displayTextAfterClick).toBeInTheDocument();
  });
});
