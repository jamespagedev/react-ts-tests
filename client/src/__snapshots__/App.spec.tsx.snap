// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultButton matches snapshot 1`] = `
<div>
  <div
    class="app"
  >
    <h3>
      strParentState: "
      "
    </h3>
    <button>
      Click Me
    </button>
  </div>
</div>
`;
